### PR TITLE
examples/gcoap_block_server: add gcoap block server example

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This adds the [gcoap block server](https://github.com/kb2ma/riot-apps/tree/kb2ma-master/gcoap-block-server) example to mainline RIOT.

It existed as an external application that is hard to discover and failed to compile with the latest upstream RIOT.

### Testing procedure

```
% aiocoap-client -m GET "coap://[fe80::7837:fcff:fe7d:1aaf%tapbr0]/riot/ver"  
This is RIOT (Version: 2022.04-devel-911-gdd8b3f) running on a native board with a native MCU.
```

### Issues/PRs references

https://github.com/RIOT-OS/RIOT/pull/17544#issuecomment-1074985169

### CODE REVIEW

File: examples/gcoap_block_server/gcoap_block.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    if (credman_add(&credential) < 0) {
+        puts("gcoap_cli: unable to add credential");
+        return;
+    }
+
+    /* tell gcoap with tag to use */
+    gcoap_set_credential_tag(SOCK_DTLS_GCOAP_TAG);
+    /* start gcoap server */
+    gcoap_init();
+#endif
```
Comment (kfessel) [inline examples/gcoap_block_server/gcoap_block.c:218]: since there is another example for DTLS we might want to remove that from this one

Comment (benpicco) [reply examples/gcoap_block_server/gcoap_block.c:218]: Isn't the other example for a DTLS *client*?

Comment (kfessel) [reply examples/gcoap_block_server/gcoap_block.c:218]: But this one seem to be a bit more flexible than the one in gcoap_dtls.

Comment (kfessel) [reply examples/gcoap_block_server/gcoap_block.c:218]: > Isn't the other example for a DTLS _client_?

no it is server and client

**Comment by benpicco** [comment]
Some context: in #17544 I want to refactor the nanocoap API and was happy to have found a solution that allowed me not to also to the same changes to GCoap.

Then @kfessel pointed out that this would break gcoap blockwise response. I was confused as I tested all gcoap applications and they worked fine - in particular, gcoap would not do blockwise responses to begin with.

Then @kfessel pointed me to this example that is linked in the documentation. It does not compile anymore with upstream RIOT, but was made to work easily enough. 

So the idea is to have this in tree to have something to test against when doing refactoring.

**Comment by miri64** [comment]
Ok, yeah, then better have a working example than no example at all ^^

File: examples/gcoap_block_server/gcoap_block.c (hunk 1 above)
Comment (benpicco) [reply examples/gcoap_block_server/gcoap_block.c]: I removed DTLS code to keep things simple.

File: examples/gcoap_block_server/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+This application provides examples of CoAP server handling for Block requests.
```
Comment (kfessel) [inline examples/gcoap_block_server/README.md:1]:
```suggestion
This application provides examples of CoAP server handling for Block requests, build with gcoap using nanocoap block handling functions.

This show the current status and is very likely to change with improvements made to gcoap and nanocoap and their API, see example/nanocoap_server for comparison.
```

**Review by kfessel — APPROVED** [review]
I think we should get this in to ensure these interaction gcoap <-> nanocoap is allways kept in mind.

**Pull request merged**